mesonbuild/modules/unstable_rust.py

```diff
@@ -203,6 +203,8 @@ def bindgen(self, state: 'ModuleState', args: T.List, kwargs: 'FuncBindgen') ->
         name: str
         if isinstance(header, File):
             name = header.fname
+        elif isinstance(header, (BuildTarget, BothLibraries, ExtractedObjects)):
+            raise InterpreterException('bindgen source file must be a C header, not an object or build target')
```

Comment:
Why was this added? It seems unsafe to put this into a stable release?

Reply:
Actually, this whole commit seems to be unrelated to the PR?

Reply:
It's in the commit message (not your fault, GitHub review is horrible):

> This also fixes a mypy failure from the next patch, since ExtractedObjects does not have a zero-argument get_outputs method.

Reply:
Okay. So I'm wondering, is it possible that someone is doing this nonsensical thing in their build files and it accidentally works somehow right now, and it will break if this goes into a stable release?

Reply:
In other words, ExtractedObjects as in the new testcase would have caused a Python backtrace, while BuildTarget and BothLibraries simply do not make sense in this position (I tried quickly `bindgen /bin/ls` and bindgen panics very soon).

Reply:
Okay, thanks :)